pytorch_lightning/core/lightning.py

```diff
@@ -221,9 +221,6 @@ def training_step(self, batch, batch_idx, hiddens):
                         "hiddens": hiddens  # remember to detach() this
                     }
 
-            You can also return a -1 instead of a dict to stop the current loop. This is useful
```

Comment:
so this doesn't work? maybe we should add back this functionality if it got removed?

Reply:
I don't think it does? Let's trace it:

training_step is only called, I think, in training_forward, which will at best return it (training_loop.py 634)

then training_forward gets called by optimizer_closure (training_loop.py 525) 
```
                    with self.profiler.profile('model_forward'):
                        output = self.training_forward(
                            split_batch, batch_idx, opt_idx, self.hiddens)

                    closure_loss = output[0]
                    progress_bar_metrics = output[1]
                    log_metrics = output[2]
                    callback_metrics = output[3]
                    self.hiddens = output[4]
```

looks a lot to me like that will just blow up if you try to return -1. 

even if not, the only thing that comes back from optimizer_closure is the loss, no kind of status or anything.

Now, this all lives inside run_training_batch, which is the thing that needs to return a -1 as its first output. And it looks to me like it's hard-coded to return 0 except in a few cases where it's hard coded to return -1 (one of which I've added to the docs in this PR)

I could be wrong but i'd be surprised.

As to whether or not to add it back, it looks to me like that would be pretty hard, and the on_batch_start solution really isn't too bad

Reply:
ok, and should your functionality also be added to batch_end? 
In any case I think a test would be good that this doesn't get lost in the future like the -1 in training_step :)

Reply:
great idea. i don't have the time to figure that out right now, maybe you can pick up the torch?

Reply:
@awaelchli do you have a test in mind?